version: "3.8"

services:
  scipy-notebook:
    image: jupyter/scipy-notebook:latest
    command:
      - jupyter
      - notebook
      - --NotebookApp.allow_password_change=False
      - --NotebookApp.password={{ secrets['users'][inventory_hostname_short]['hashed_pass'] | replace('$','$$') }}
    environment:
      - GRANT_SUDO=no
      - RESTARTABLE=yes
      - CHOWN_HOME=yes
    networks:
      - web
    volumes:
      - "{{ data_dir.path }}:/home/jovyan/work"
    restart: unless-stopped
    labels:
      - traefik.enable=true
      - traefik.http.routers.{{ role_name }}.rule=Host(`{{ inventory_hostname_short }}.jupyter.ber.lab.alpaka.space`)
      - traefik.http.routers.{{ role_name }}.entrypoints=https
      - traefik.http.routers.{{ role_name }}.tls=true
      - traefik.http.routers.{{ role_name }}.tls.certresolver=letsencrypt
      - traefik.http.services.{{ role_name }}.loadbalancer.server.port=8888
      - com.centurylinklabs.watchtower.enable=true

networks:
  web:
    external:
      name: traefik_web
